Library/Homebrew/cmd/update.rb

```diff
@@ -411,10 +411,33 @@ def select_formula(key)
 
   def dump_formula_report(key, title)
     formula = select_formula(key)
-    formula.map! { |oldname, newname| "#{oldname} -> #{newname}" } if key == :R
     unless formula.empty?
+      # Determine list item indices of installed formulae.
+      formula_installed = formula.each_index.select do |index|
+        name, newname = formula[index]
+        installed?(name) || (newname && installed?(newname))
+      end
+
+      # Format list items of renamed formulae.
+      if key == :R
+        formula.map! { |oldname, newname| "#{oldname} -> #{newname}" }
+      end
+
+      # Append suffix '(installed)' to list items of installed formulae.
+      formula_installed.each do |index|
+        formula[index] += " (installed)"
+      end
+
+      # Fetch list items of installed formulae for highlighting.
+      formula_installed = formula.values_at(*formula_installed)
+
+      # Dump formula list.
       ohai title
-      puts_columns formula
+      puts_columns(formula, formula_installed)
     end
   end
+
+  def installed?(formula)
+    (HOMEBREW_CELLAR / formula.split("/", 3).last).directory?
```

Comment:
My intention was to make the behavior of `split` a bit more robust and I have to admit, that I'm still having a hard time leaving out spaces around binary operators (though here it's a operation on path). But I'll happily change this as suggested.